lib/librato/collector/counter_cache.rb

```diff
@@ -49,16 +53,13 @@ def flush_to(queue, opts={})
         @lock.synchronize do
           # work off of a duplicate data set so we block for
           # as little time as possible
-          counts = @cache.dup
+          # requires a deep copy of data set
+          counts = JSON.parse(@cache.dup.to_json, symbolize_names: true)
           reset_cache unless opts[:preserve]
         end
         counts.each do |metric, value|
-          metric, source = metric.split(SEPARATOR)
-          if source
-            queue.add metric => {value: value, source: source}
-          else
-            queue.add metric => value
-          end
+          metric = metric.to_s.split(SEPARATOR).first
+          queue.add metric => value
```

Comment:
Assuming the tags are already present in `value` via the JSON parse above? If so this local should probably be renamed to make that clearer - the first time I read this I assumed they were being lost.